.github/workflows/doc-build.yml

```diff
@@ -0,0 +1,111 @@
+name: Documentation Build and Deploy
+
+on:
+  push:
+    branchs:
+      - arc-dev
+  pull_request:
+    paths:
+    - 'doc/**'
+    - '**.rst'
+    - '.github/workflows/doc-build.yml'
+
+jobs:
+  doc-build-html:
+    name: "Documentation Build (HTML)"
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+
+    - name: install-pkgs
+      run: |
+        sudo apt-get install -y doxygen
+
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ runner.os }}-doc
+
+    - name: install-pip
+      run: |
+        sudo pip3 install -U setuptools wheel pip
+        pip3 install breathe
```

Comment:
@wangnuannuan, could you please create requirements.txt in doc directory with required package versions and call 'pip3 install -r requirements.txt' instead?